### PR TITLE
Transfuser TT-CNN Functional

### PR DESCRIPTION
**Pull request by ign-saurav**
### Overview
This PR integrates Transfuser model support into the Tenstorrent codebase(`tt-metal`), enabling transformer based multimodel sensor fusion on TTNN devices using TT-CNN architecture.

Input Resolutions

- Image (RGB, NCHW): 1 × 3 × 160 × 704

   - Three front cameras are cropped & concatenated to 160×704.

- LiDAR (BEV, NCHW): 1 × 1 × 256 × 256

  - BEV grid input at 256×256.

### Source
- [Transfuser](https://github.com/autonomousvision/transfuser)

### TransFuser – Model Tree

transfuser
├── image_encoder/ # RegNet (RGB: 1×3×160×704)
│ ├── stem
│ └── stages s1–s4 # RegNet bottleneck blocks
│
├── lidar_encoder/ # RegNet (BEV: 1×1×256×256)
│ ├── stem
│ └── stages s1–s4
│
├── projection/
│ ├── img_proj # to transformer dim D
│ └── lidar_proj
│
├── fusion_transformer/
│ ├── encoder_blocks (L) # MHSA + MLP
│ └── cross_attention # img ↔ lidar
│
├── fusion_head/
│ └── upsample_path # upsample + (depthwise) conv
│
└── heads/
├── semantic_head # segmentation logits
├── center_head # instance center heatmap
└── offset_head # instance offsets

#### Hardware: Wormhole n150
seed=42

| Backbone Output | PCC Score |
|-----------------|-----------|
| p2 (FPN)        | 0.9995 |
| p3 (FPN)        | 0.9996 |
| p4 (FPN)        | 0.9994 |
| p5 (FPN)        | 0.9992 |
| Image Grid      | 0.9993 |
| Fused Features  | 0.9950 |

### Per-Head PCC Breakdown

| Component                | PCC Score |
|--------------------------|-----------|
| Fused Features           | 0.9947 |
| Predicted Waypoints      | 0.9999 |
| Feature Map              | 0.9997 |
| Detection Head (Overall) | 0.9973 |
| Bounding Boxes (Boxes)   | 0.9079 |

### Detection Head – Per-Head PCC Breakdown

| Head Component | PCC Score |
|---------------|-----------|
| Heatmap       | 0.9973 |
| WH (Width/Height) | 0.9992 |
| Offset        | 0.9979 |
| Yaw Class     | 0.9997 |
| Yaw Residual  | 0.9932 |
| Velocity      | 0.9993 |
| Brake         | 0.9998 |

### Tests
Full network test (with real weights and real input from image and lidar):
`pytest models/experimental/transfuser/tests/pcc/test_lidar_center_net.py`

### Demo Test:
`python models/experimental/transfuser/demo/lidar_center_net_demo.py`

### Perf test
`pytest models/experimental/transfuser/tests/perf/test_perf.py`

### Performance:

Device Performance FPS : 28.08
E2E FPS (2CQ + trace): 25.96

### Layers running on torch

- Other than transfuser rest of lidarcenternet is running on torch

### Checklist

- [ ] [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml?query=branch:ign/transfuser-ttcnn)
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:ign/transfuser-ttcnn)
- [ ] [cpp-unit-tests](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:ign/transfuser-ttcnn)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by dvartaniansTT — COMMENTED** [review]
LGTM, just need 1-2 weeks to test all pytests for all the PRs locally in order to approve. also requires rebasing to metal's latest main for us and testing from there. 
@ign-saurav is the reported FPS end to end or device? pls include both in the model README

**Comment by ign-febin** [comment]
> LGTM, just need 1-2 weeks to test all pytests for all the PRs locally in order to approve. also requires rebasing to metal's latest main for us and testing from there. @ign-saurav is the reported FPS end to end or device? pls include both in the model README

Updated the Readme file and description, added E2E Performance FPS (25.96) and Device Performance (28.08)
